parsec/core/local_storage.py

```
@@ -261,3 +263,106 @@ def create_cursor(self, entry_id: EntryID) -> FileDescriptor:
+        assert task is None or task == hazmat.current_task()
+
+    # Manifest interface
+
+    def get_base_manifest(self, entry_id: EntryID) -> RemoteManifest:
+        """Raises: LocalStorageMissingError"""
+        assert isinstance(entry_id, EntryID)
+        try:
+            return self.base_manifest_cache[entry_id]
+        except KeyError:
+            raise LocalStorageMissingError(entry_id)
+
+    def get_manifest(self, entry_id: EntryID) -> LocalManifest:
+        """Raises: LocalStorageMissingError"""
+        assert isinstance(entry_id, EntryID)
+        try:
+            return self.local_manifest_cache[entry_id]
+        except KeyError:
+            raise LocalStorageMissingError
+
+    def set_base_manifest(
+        self, entry_id: EntryID, manifest: RemoteManifest
+    ) -> None:  # initially for clean
+        assert isinstance(entry_id, EntryID)
+        self._check_lock_status(entry_id)
+        self.base_manifest_cache[entry_id] = manifest
+
+    def set_manifest(
+        self, entry_id: EntryID, manifest: LocalManifest
+    ) -> None:  # initially for dirty
+        assert isinstance(entry_id, EntryID)
+        assert manifest.author == self.device_id
+        self._check_lock_status(entry_id)
+        self.local_manifest_cache[entry_id] = manifest
+
+    def clear_manifest(self, entry_id: EntryID) -> None:
+        assert isinstance(entry_id, EntryID)
+        self._check_lock_status(entry_id)
+        self.base_manifest_cache.pop(entry_id, None)
+        self.local_manifest_cache.pop(entry_id, None)
+
+    # Block interface
+
+    def set_dirty_block(self, block_id: BlockID, block: bytes) -> None:
+        # should not return anything? no sense to write dirty blocks from ltimestamped? raise error?
+        return
+
+    def clear_block(self, block_id: BlockID) -> None:
+        # better not uncache a maybe used block by the actual workspace than to have to reload it?
+        return
```

I would also raise an error here.